compiler/rustc_borrowck/src/diagnostics/conflict_errors.rs

```diff
@@ -446,10 +448,77 @@ impl<'cx, 'tcx> MirBorrowckCtxt<'cx, 'tcx> {
                 }
             }
         }
+
         err.span_label(decl_span, "binding declared here but left uninitialized");
+        if show_assign_sugg {
+            struct LetVisitor {
+                decl_span: Span,
+                ty_span: Option<Span>,
+            }
+
+            impl<'v> Visitor<'v> for LetVisitor {
+                fn visit_stmt(&mut self, ex: &'v hir::Stmt<'v>) {
+                    if self.ty_span.is_some() {
+                        return;
+                    }
+                    if let hir::StmtKind::Local(hir::Local {
+                            span, init: None, ty: Some(ty), ..
+                        }) = &ex.kind && span.contains(self.decl_span) {
+                            self.ty_span = Some(ty.span);
+                    }
+                    hir::intravisit::walk_stmt(self, ex);
+                }
+            }
+
+            let mut visitor = LetVisitor { decl_span, ty_span: None };
+            visitor.visit_body(&body);
+            if let Some(ty_span) = visitor.ty_span {
```

Comment:
Hm, seems quite unfortunate that we only handle the bindings that have a specified type. As such code like this isn’t really covered:

```
fn apple(_: u32) { }

fn banana() {
    let chaenomeles;
    apple(chaenomeles);
}
```

Not saying though that this needs to be implemented here and now, but perhaps a FIXME comment or an issue for this would be warranted.

Reply:
Fixed the code to support this scenario now, the type infer know it's a `u32`, so here we give this suggestion:

```diff
error[E0381]: used binding `chaenomeles` isn't initialized
 --> bug.rs:5:11
  |
4 |     let chaenomeles;
  |         ----------- binding declared here but left uninitialized
5 |     apple(chaenomeles);
  |           ^^^^^^^^^^^ `chaenomeles` used here but it isn't initialized
  |
help: consider assigning a value
  |
4 |     let chaenomeles = 0;
  |                     +++
```